connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -109,6 +109,7 @@ message Join {
   Relation right = 2;
   Expression join_condition = 3;
   JoinType join_type = 4;
+  repeated string using_columns = 5;
```

Comment:
Can join_condition and using_columns co-exist?

Reply:
Based on the current Catalyst implementation. My read is yes.

My understanding is current Catalyst implementation does not support `USING ON col_name,...` in the join_condition. It is a separate code path. Is it true?

Reply:
Looking at the API layer (DF and SQL), I don't think they can co-exist. E.g. the parser rule is
```
joinCriteria
    : ON booleanExpression
    | USING identifierList
    ;
```

Reply:
I see. To match with our API (but not implementation), I added a check to make sure only one of these two  will be set.